doc/eventhandler/requestmangler.rst

```diff
@@ -0,0 +1,90 @@
+.. _requestmanglerhandler:
+
+RequestMangler Handler Module
+-----------------------------
+
+.. index:: RequestMangler, Handler Modules
+
+The RequestMangler is a special handler module, that can modify
+the request parameters of an HTTP request.
+This way privacyIDEA can change the data that is processed within the request.
+
+Usually this handler is used in the **pre** location. However there might be occasions
+when you want to modify paramters only *before* passing them to the next handler. In this case
+you can also use the RequestMangler handler in the **post** location.
+
+Possible Actions
+~~~~~~~~~~~~~~~~
+
+delete
+......
+
+This action simply deletes the given parameter from the request.
+
+E.g. you could in certain cases delete the ``transaction_id`` from a
+``/validate/check`` request. This way you would render challenge response inactive.
+
+add
+...
+
+This action is used to add or modify additional request parameters.
+
+You can set a parameter with the value or substrings of another paramter.
+
+This is why this action takes the additional options *value*, *match_parameter* and *match_pattern*.
+
+If you simply want to set a parameter to a fixed value you only need the options:
+
+* *parameter*: as the name of the parameter you want to set and
+* *value*: to set to a fixed value.
+
+If you can to set a parameter based on the value of another parameter, you can use the regex notation
+**()** and the python string formatting tags **{0}**, **{1}**.
+
+**Example 1**
+
+To set the realm based on the user parameter:
```

Comment:
I think we should use a double colon like *... user parameter::* here to avoid weird formatting issues with the match pattern below.

Reply:
done in a8f1447c